sdks/java/io/amazon-web-services2/src/main/java/org/apache/beam/sdk/io/aws2/kinesis/ShardReadersPool.java

```diff
@@ -255,6 +256,16 @@ Instant getLatestRecordTimestamp() {
     return getMinTimestamp(ShardRecordsIterator::getLatestRecordTimestamp);
   }
 
+  Instant getEventTimestamp(KinesisRecord record) {
+    ImmutableList<ShardRecordsIterator> shardRecordsIterators =
+        shardIteratorsMap.get().values().asList();
+
+    if (shardRecordsIterators.isEmpty()) {
+      return null;
+    }
+    return shardRecordsIterators.get(0).getEventTime(record);
```

Comment:
Same question here, is it valid to ask for the event time from any shard record iterator?

Reply:
We get watermark also from `ShardRecordsIterator#getWatermark` so it seems logical that event timestamp is also fetched from `ShardRecordsIterator#getEventTime` which internally returns `watermarkPolicy.getTimestamp`.

Reply:
SGTM